docs/modules/ROOT/pages/accounts.adoc

```diff
@@ -98,3 +98,230 @@ But if you don't have an account to invoke it, you will probably want to use the
+The flag enables custom account variants.
+
+```[,bash]
+$ sncast \
+  --url http://127.0.0.1:5050 \
+  account create \
+  --name my-custom-account \
+  --class-hash 0x123456...
+```
+
+This command will output the precomputed contract address and the recommended `max-fee`.
+To counterfactually deploy the account, send funds to the address and then deploy the custom account.
+
+```[,bash]
+$ sncast \
+  --url http://127.0.0.1:5050 \
+  account deploy \
+  --name my-custom-account
+```
+
+Once the account is deployed, set the `--account` flag with the custom account name to send transactions from that account.
+
+```[,bash]
+$ sncast \
+  --account my-custom-account \
+  --url http://127.0.0.1:5050 \
+  invoke \
+  --contract-address 0x123... \
+  --function "some_function" \
+  --calldata 1 2 3
+```
+
+=== EthAccountComponent
+
+:eth-account-mixin: xref:/api/account.adoc#EthAccountComponent-Embeddable-Mixin-Impl[EthAccountMixin]
+:eth-initializer: xref:/api/account.adoc#EthAccountComponent-initializer[initializer]
+:starknetjs: https://www.starknetjs.com/[StarknetJS]
+
+To create an Ethereum-flavored account, implement the {eth-account-mixin} and set up the constructor to initialize the contract.
+Since this is an EthAccount, the {eth-initializer} expects the `EthPublicKey` type (alias for `Secp256k1Point`) to store as the account's public key.
+The contract also requires the `Secp256K1Impl` implementation in order to serialize and deserialize the `EthPublicKey` type.
+A basic Ethereum-flavored account contract looks like this:
+
+```[,cairo]
+#[starknet::contract(account)]
+mod MyEthAccount {
+    use openzeppelin::account::EthAccountComponent;
+    use openzeppelin::account::interface::EthPublicKey;
+    use openzeppelin::account::utils::secp256k1::Secp256k1PointSerde;
+    use openzeppelin::introspection::src5::SRC5Component;
+
+    component!(path: EthAccountComponent, storage: eth_account, event: EthAccountEvent);
+    component!(path: SRC5Component, storage: src5, event: SRC5Event);
+
+    // EthAccount Mixin
+    #[abi(embed_v0)]
+    impl EthAccountMixinImpl =
+        EthAccountComponent::EthAccountMixinImpl<ContractState>;
+    impl EthAccountInternalImpl = EthAccountComponent::InternalImpl<ContractState>;
+
+    #[storage]
+    struct Storage {
+        #[substorage(v0)]
+        eth_account: EthAccountComponent::Storage,
+        #[substorage(v0)]
+        src5: SRC5Component::Storage
+    }
+
+    #[event]
+    #[derive(Drop, starknet::Event)]
+    enum Event {
+        #[flat]
+        EthAccountEvent: EthAccountComponent::Event,
+        #[flat]
+        SRC5Event: SRC5Component::Event
+    }
+
+    #[constructor]
+    fn constructor(ref self: ContractState, public_key: EthPublicKey) {
+        self.eth_account.initializer(public_key);
+    }
+}
+```
+
+Special tooling is required in order to deploy and send transactions with an Ethereum-flavored account contract.
+The following examples utilize the {starknetjs} library.
```

Comment:
We should add a NOTE with the version of starknet-js used in the examples (similar to the sncast NOTE).